From 2b4005d72d3393933a7914be102ea65505c536cc Mon Sep 17 00:00:00 2001
From: "Maxin B. John" <maxin.john@intel.com>
Date: Thu, 21 Jul 2016 11:53:31 +0300
Subject: [PATCH] configure: Add option to enable/disable polkit

Changes the configure behaviour from autodetecting the polkit by default
to having an option to disable it explicitly

Upstream-Status: Pending

Signed-off-by: Maxin B. John <maxin.john@intel.com>
---
 configure.ac | 8 ++++++++
 1 file changed, 8 insertions(+)

diff --git a/configure.ac b/configure.ac
index 2246d5a..3d3fe0f 100644
--- a/configure.ac
+++ b/configure.ac
@@ -104,10 +104,18 @@ PKG_CHECK_MODULES(GTK,
                   [gtk+-3.0 >= gtk_required_version],
                   [have_gtk=yes],
                   [have_gtk=no])
+AC_ARG_ENABLE([polkit],
+                   AS_HELP_STRING([--disable-polkit], [Do not use Polkit]),
+                   [enable_polkit="$enableval"], [enable_polkit="yes"])
+
+AS_IF([test "x$enable_polkit" = "xyes"], [
 PKG_CHECK_MODULES(POLKIT,
                   [polkit-gobject-1],
                   [have_polkit=yes],
                   [have_polkit=no])
+                                         ])
+AM_CONDITIONAL([HAVE_POLKIT], [test "x$enable_polkit" = "xyes"])
+
 PKG_CHECK_MODULES(SYSTEMD,
                   [libsystemd >= systemd_required_version],
                   [have_systemd=yes],
-- 
2.4.0

